e2e/framework/provisioning/runner_linux.go

```diff
@@ -1,25 +1,57 @@
 package provisioning
 
 import (
+	"context"
+	"fmt"
+	"log"
 	"os/exec"
 	"strings"
 	"testing"
+	"time"
+
+	"github.com/pkg/errors"
 )
 
 // LinuxRunner is a ProvisioningRunner that runs on the executing host only.
 // The Nomad configurations used with this runner will need to avoid port
 // conflicts!
-type LinuxRunner struct{}
+//
+// Must call Open before other methods.
+type LinuxRunner struct {
+	// populated on Open.
+	t *testing.T
+}
 
-func (runner *LinuxRunner) Open(_ *testing.T) error { return nil }
+// Open sets up the LinuxRunner to run using t as a logging mechanism.
+func (runner *LinuxRunner) Open(t *testing.T) error {
+	runner.t = t
+	return nil
+}
 
+func parseCommand(command string) (string, []string) {
+	fields := strings.Fields(strings.TrimSpace(command))
+	if len(fields) == 1 {
+		return fields[0], nil
+	}
+	return fields[0], fields[1:]
+}
+
+// Run the script (including any arguments)
 func (runner *LinuxRunner) Run(script string) error {
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Minute)
```

Comment:
Nothing to do, just thinking out loud: We probably want to be able to pass a context to the provisioner or make this timeout configurable at some point.